homeassistant/bootstrap.py

```diff
@@ -96,6 +99,25 @@ def _setup_component(hass, domain, config):
                 domain, ", ".join(missing_deps))
             return False
 
+        if hasattr(component, 'CONFIG_SCHEMA'):
+            try:
+                config = component.CONFIG_SCHEMA(config)
+            except vol.MultipleInvalid as ex:
+                _LOGGER.error('Invalid config for [%s]: %s', domain, ex)
+                return False
+
+        if hasattr(component, 'PLATFORM_SCHEMA'):
+            platforms = []
+            for _, platform in config_per_platform(config, domain):
+                try:
+                    platforms.append(component.PLATFORM_SCHEMA(platform))
+                except vol.MultipleInvalid as ex:
+                    _LOGGER.error('Invalid platform config for [%s]: %s. %s',
+                                  domain, ex, platform)
+                    return False
+
+            config = {domain: platforms}
```

Comment:
What is the point of the local config variable here, it isn't used in this function and it clobbers what was assigned to  config (and not used) at line 104 above.